pytensor/tensor/rewriting/elemwise.py

```diff
@@ -380,6 +380,99 @@ def is_dimshuffle_useless(new_order, input):
     return is_useless
 
 
+@node_rewriter([Elemwise])
+def local_elemwise_lift_scalars(fgraph, node):
+    op = node.op
+
+    if not isinstance(op, Elemwise):
+        return False
+
+    if not all(input.ndim == 0 for input in node.inputs):
+        return False
+
+    scalars = [aes.as_scalar(input) for input in node.inputs]
+
+    # TODO Something like
+    # copy_stack_trace(node.outputs[0], new_res)
+    return [as_tensor_variable(out) for out in op.scalar_op.make_node(*scalars).outputs]
+
+
+compile.optdb["specialize"].register(
```

Comment:
specialize is optional, if Numba will fail without this rewrite we should add a new non optional rewrite phase at the end.

Reply:
I think we should probably make it so that elemwise doesn't fail in numba either way.
To be honest, I'm having a hard time seeing how we could expect everything to work nicely if users pick and choose rewrites. I think that is a testing nightmare...

Reply:
I'll have to rething the phases a bit for sure though. Maybe this actually belongs in "uncanonicalize" or so? Or we could have a new phase "scalarize"...

Reply:
I moved it to a new "scalarize" phase